File: plugins/vitest-agent-observability/src/judges/adapters/openai.ts
/**
 * OpenAI-compatible LLM adapter using fetch (no SDK dependency).
 *
 * Sync target:
 *   plugins/pytest-agent-observability/src/pytest_agent_observability/judges/adapters/openai.py
 */

import type { LLMClient } from "../runner.js";

export function openaiAdapter({
  apiKey,
  model = "gpt-4o-mini",
  baseUrl,
}: {
  apiKey: string;
  model?: string;
  baseUrl?: string;
}): LLMClient {
  if (!apiKey) throw new Error("openaiAdapter: apiKey is required");

  const url =
    (baseUrl ?? "https://api.openai.com").replace(/\/$/, "") +
    "/v1/chat/completions";

  return {
    async evaluate(prompt: string): Promise<string> {
      const resp = await fetch(url, {
        method: "POST",
        headers: {
          Authorization: `Bearer ${apiKey}`,
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          model,
          temperature: 0,
          response_format: { type: "json_object" },
          messages: [{ role: "user", content: prompt }],
        }),
      });

      if (!resp.ok) {
        const text = await resp.text().catch(() => "");
        throw new Error(`openaiAdapter: HTTP ${resp.status} — ${text}`);
      }

      const data = (await resp.json()) as any;
      return data.choices[0].message.content as string;
    },
  };
}

File: plugins/pytest-agent-observability/src/pytest_agent_observability/judges/adapters/openai.py
"""OpenAI-compatible LLM adapter using httpx (already a project dependency)."""

from __future__ import annotations

from typing import Any

import httpx

from ..types import LLMClient


def openai_adapter(
    api_key: str,
    model: str = "gpt-4o-mini",
    base_url: str | None = None,
) -> LLMClient:
    """Return an LLMClient that calls the OpenAI chat completions endpoint.

    Uses httpx (already a pytest-agent-observability dependency).
    Sets temperature=0 and response_format=json_object for deterministic JSON output.
    """
    url = (base_url or "https://api.openai.com").rstrip("/") + "/v1/chat/completions"

    class _OpenAIClient:
        def evaluate(self, prompt: str) -> str:
            payload: dict[str, Any] = {
                "model": model,
                "temperature": 0,
                "response_format": {"type": "json_object"},
                "messages": [{"role": "user", "content": prompt}],
            }
            resp = httpx.post(
                url,
                json=payload,
                headers={
                    "Authorization": f"Bearer {api_key}",
                    "Content-Type": "application/json",
                },
                timeout=60.0,
            )
            resp.raise_for_status()
            data = resp.json()
            return data["choices"][0]["message"]["content"]

    return _OpenAIClient()
